### PR TITLE
Upgrade dependencies and set minimal PHP version to 7.4

### PR DESCRIPTION
**Pull request by cedric-anne**
Purpose of this pull request is to be able to use `sabre/dav` in combination wih latest versions of other `sabre` components into a same project.

I aligned PHP requirements (PHP 7.4 -> 8.2), but maybe we can drop here support of PHP 7.4.

Before releasing a new 5.0 version, code should probably modernized (for instance by adding return types, parameters type hinting, ...), but this would be part of another PR.

### CODE REVIEW

**Comment by cedric-anne** [comment]
Not sure about hox to fix the error shown in test suite. If I add a return type to the method `Sabre\DAV\Xml\Property\Complex::xmlDeserialize()`, it will result in following error: `TypeError: Sabre\DAV\Xml\Property\Complex::xmlDeserialize(): Return value must be of type Sabre\Xml\Element\XmlFragment, string returned`

File: lib/CalDAV/Backend/PDO.php [hunk 1]
```diff
@@ -1462,7 +1462,7 @@ public function getInvites($calendarId)
             $result[] = new Sharee([
                 'href' => isset($row['share_href']) ? $row['share_href'] : \Sabre\HTTP\encodePath($row['principaluri']),
                 'access' => (int) $row['access'],
-                /// Everyone is always immediately accepted, for now.
+                // / Everyone is always immediately accepted, for now.
```
Comment (AnaelMobilia) [inline lib/CalDAV/Backend/PDO.php:1465]: Third slash should be removed

**Comment by cedric-anne** [comment]
@phil-davis 

I think I will not be able to finish this task. Main problem here is that return types of `Sabre\DAV\Xml\Property\Complex::xmlDeserialize()` and `Sabre\DAV\Xml\Property\ResourceType::xmlDeserialize()` are not compatible with the definition in extended class. Changing it has many side effects and I do not know how to manage it.

**Comment by phil-davis** [comment]
I need to keep moving with my voluntary work on https://github.com/sabre-io/vobject/pull/591 and get a major version published there. Then all the sabre-io/dav dependencies will have new major versions that support only PHP 7.4 and up. Then "we" (whoever can provide effort) can work on the same for `sabre-io/dav`

**Comment by cedric-anne** [comment]
I wanted to help on this point, but I do not know the project well enough to carry out this first step.

I close this PR, maybe I will be able to help better when someone who knows the project better takes over the job.

**Pull request closed without merging**